### ISSUE TITLE
[Bug]: iOS native plugin extern function name conflict

### ISSUE THREAD

**Issue by AFB-MORA** [OneSignal/OneSignal-Unity-SDK#663]
### What happened?

Native plugin extern function name "_logout()" conflict with other plugin's (Branch.io) function name "_logout()". General function names like "_logout()" should not be used. You can rename general names with "oneSignal" prefix like Facebook's "IOSFBLogOut();" function.

This issue is not specific to Branch.io. If I write my own custom plugin with "_logout()" or "_initialize(string)" functions, I can still see same error. These names are too general and cannot be used.

### Steps to reproduce?

```Markdown
1. Add OneSignal
2. Add Branch.io
3. Build iOS project
4. Archive Xcode project
5. You see duplicate symbols error
```

### What did you expect to happen?

Build Xcode project successfully.

### Unity version

2022.3.10

### OneSignal Unity SDK version

5.0.5

### Platform

iOS

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by shepherd-l** [comment]
Could you provide more context on what you are trying to do?
Could you also share the error you are seeing

**Comment by AFB-MORA** [comment]
I want to use Branch.io (attribution tool) along with OneSignal. When I try to build iOS in Xcode, I see the error:

```sh
duplicate symbol '__logout' in:
    [...]/Library/Developer/Xcode/DerivedData/Unity-iPhone-djwsngoaxtltiwbqgubrkoxojwfv/Build/Intermediates.noindex/ArchiveIntermediates/Unity-iPhone/IntermediateBuildFilesPath/Unity-iPhone.build/Release-iphoneos/UnityFramework.build/Objects-normal/arm64/BranchiOSWrapper.o
    [...]/Library/Developer/Xcode/DerivedData/Unity-iPhone-djwsngoaxtltiwbqgubrkoxojwfv/Build/Intermediates.noindex/ArchiveIntermediates/Unity-iPhone/IntermediateBuildFilesPath/Unity-iPhone.build/Release-iphoneos/UnityFramework.build/Objects-normal/arm64/OneSignalUnityBridge.o
ld: 1 duplicate symbol for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I think the problem is how static extern functions work. OneSignal sdk has its own native "_logout()" implementation, and Branch sdk has its own native "_logout()" implementation. So when I build, I think there is two compiled static "_logout()" functions. 

The workaround I found is renaming "_logout()" function in Branch sdk to "_branchLogout()" both in native code and C# code.

I think general names like "_logout()" should not be used in static extern functions to prevent conflicts. For examples Facebook sdk uses "IOSFBLogOut()" for its own logout implementation.

**Comment by shepherd-l** [comment]
Thanks for the clarity and detailed post
I’ll prefix our Objective-C methods to avoid conflicts.

**Comment by shepherd-l** [comment]
This should be fixed in the latest [5.0.6 release](https://github.com/OneSignal/OneSignal-Unity-SDK/releases/tag/5.0.6). Please open a new issue if you have any questions
